subsys/net/ip/net_pkt.c

```diff
@@ -1735,28 +1737,36 @@ size_t net_pkt_remaining_data(struct net_pkt *pkt)
 
 int net_pkt_update_length(struct net_pkt *pkt, size_t length)
 {
+	size_t len = length;
 	struct net_buf *buf;
 
 	for (buf = pkt->buffer; buf; buf = buf->frags) {
-		if (buf->len < length) {
-			length -= buf->len;
+		if (buf->len < len) {
+			len -= buf->len;
 		} else {
-			buf->len = length;
-			length = 0;
+			buf->len = len;
+			len = 0;
 		}
 	}
 
-	return !length ? 0 : -EINVAL;
+	if (len) {
+		return -EINVAL;
+	}
+
+	pkt->length -= length;
```

Comment:
Doing this without any checks looks dangerous. What if pkt->length goes <0 and becomes a large value.

Reply:
It is unlikely to happen if, and only if, all operations are done through net_pkt API. But we cannot prevent users to mess up with net_buf I guess. So let's be conservative then yes